### PR TITLE
test: fix the auth problem in the test of set replica for system tables

### PR DESCRIPTION
**Pull request by AilinKid**
Signed-off-by: ailinkid <314806019@qq.com>

### What problem does this PR solve?

Problem Summary: fix the auth problem in the test of set replica for system tables

### What is changed and how it works?

How it Works: fix the auth problem in the test of set replica for system tables

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- fix the auth problem in the test of set replica for system tables

### CODE REVIEW

File: ddl/db_test.go [hunk 1]
```diff
@@ -5340,13 +5341,19 @@ func (s *testSerialDBSuite) TestSetTableFlashReplicaForSystemTable(c *C) {
...
 		for i := 0; i < len(rows); i++ {
 			sysTables = append(sysTables, rows[i][0].(string))
 		}
 		for _, one := range sysTables {
 			_, err := tk.Exec(fmt.Sprintf("alter table `%s` set tiflash replica 1", one))
-			c.Assert(err.Error(), Equals, "[ddl:8200]ALTER table replica for tables in system database is currently unsupported")
+			if db == "MySQL" {
+				c.Assert(err.Error(), Equals, "[ddl:8200]ALTER table replica for tables in system database is currently unsupported")
+			} else {
+				c.Assert(err.Error(), Equals, fmt.Sprintf("[planner:1142]ALTER command denied to user 'root'@'%%' for table '%s'", strings.ToLower(one)))
```
Comment (djshow832) [inline ddl/db_test.go:5354]: Why is the error message different in other system databases?

File: ddl/db_test.go [hunk 2]
```diff
@@ -5340,13 +5341,19 @@ func (s *testSerialDBSuite) TestSetTableFlashReplicaForSystemTable(c *C) {
...
 		tk.MustExec("use " + db)
+		tk.Se.Auth(&auth.UserIdentity{Username: "root", Hostname: "%"}, nil, nil)
 		rows := tk.MustQuery("show tables").Rows()
 		for i := 0; i < len(rows); i++ {
 			sysTables = append(sysTables, rows[i][0].(string))
 		}
 		for _, one := range sysTables {
 			_, err := tk.Exec(fmt.Sprintf("alter table `%s` set tiflash replica 1", one))
-			c.Assert(err.Error(), Equals, "[ddl:8200]ALTER table replica for tables in system database is currently unsupported")
+			if db == "MySQL" {
```
Comment (djshow832) [inline ddl/db_test.go:5351]: Is it `MySQL`, not `mysql` or `MYSQL`?

Comment (AilinKid) [reply ddl/db_test.go:5351]: i listed it `memOrSysDB := []string{"MySQL", "INFORMATION_SCHEMA", "PERFORMANCE_SCHEMA", "METRICS_SCHEMA"}` above

File: ddl/db_test.go (hunk 1 above)
Comment (AilinKid) [reply ddl/db_test.go:5354]: user can not modify the any table in this tree database except mysql

Comment (djshow832) [reply ddl/db_test.go:5354]: So they can modify system tables like `information_schema.tables` and memory tables like `information_schema.statements_summary`?

**Review by djshow832 — COMMENTED** [review]
~~No test cases.~~

File: ddl/db_test.go (hunk 1 above)
Comment (AilinKid) [reply ddl/db_test.go:5354]: No, they don't have the privilege

Comment (zimulala) [reply ddl/db_test.go:5354]: Should MySQL return this error as well?

Comment (AilinKid) [reply ddl/db_test.go:5354]: mysql> alter table user comment "ds"; 
Query OK, 0 rows affected (0.04 sec)
Records: 0  Duplicates: 0  Warnings: 0

mysql can alter the tables in this database

**Review by zimulala — APPROVED** [review]
LGTM

**Review by djshow832 — APPROVED** [review]

**Pull request merged**